packages/store/src/StoreCore.sol

```diff
@@ -134,6 +131,11 @@ library StoreCore {
     string[] memory keyNames,
     string[] memory fieldNames
   ) internal {
+    // Verify the table ID is of type RESOURCE_TABLE
+    if (!tableId.isType(RESOURCE_TABLE)) {
```

Comment:
should this also include offchain tables?

Reply:
is it worth including a free function in `storeResourceTypes.sol` something like
```
function isTable(ResourceId resourceId) { ... }
```
that we can use here? or is the indirection too gassy?

Reply:
increases gas by 40, see https://github.com/latticexyz/mud/pull/1557/commits/ef89c43d3a9316bcfb523c752ef0c39a130f5cdd - not super significant, but still don't think it's worth the small increase in ergonomics (agree it would read nicer, sad that the Solidity compiler doesn't just optimize this away)